assets/src/edit-story/components/canvas/singleSelectionMoveable/index.js

```diff
@@ -0,0 +1,225 @@
+    latestEvent.current = pushEvent;
+  }, [pushEvent]);
+
+  useEffect(() => {
+    if (!moveable.current) {
+      return;
+    }
+    // If we have persistent event then let's use that, ensuring the targets match.
+    if (
+      latestEvent.current &&
+      targetEl.contains(latestEvent.current.target) &&
+      actionsEnabled
+    ) {
+      moveable.current.moveable.dragStart(latestEvent.current);
+    }
+    moveable.current.updateRect();
+  }, [targetEl, moveable, actionsEnabled]);
+
+  // Update moveable with whatever properties could be updated outside moveable
+  // itself.
+  useEffect(() => {
+    if (!moveable.current) {
+      return;
+    }
+    moveable.current.updateRect();
+  });
+
+  const box = getBox(selectedElement);
+  let frame = useMemo(
+    () => ({
+      translate: [0, 0],
+      rotate: box.rotationAngle,
+      resize: [0, 0],
+      updates: null,
+    }),
+    [box.rotationAngle]
+  );
+
+  const setTransformStyle = (target, newFrame = frame) => {
+    // Get the changes coming from each action type.
+    frame.translate = newFrame.translate;
+    frame.direction = newFrame.direction;
+    frame.resize = newFrame.resize;
+    frame.updates = newFrame.updates;
+    frame.rotate = newFrame.rotate;
+    target.style.transform = `translate(${frame.translate[0]}px, ${frame.translate[1]}px) rotate(${frame.rotate}deg)`;
+    if (frame.resize[0]) {
+      target.style.width = `${frame.resize[0]}px`;
+    }
+    if (frame.resize[1]) {
+      target.style.height = `${frame.resize[1]}px`;
+    }
+    pushTransform(selectedElement.id, frame);
+  };
+
+  /**
+   * Resets Moveable once the action is done, sets the initial values.
+   *
+   * @param {Object} target Target element.
+   */
+  const resetMoveable = useBatchingCallback(
+    (target) => {
+      frame.direction = [0, 0];
+      frame.translate = [0, 0];
+      frame.resize = [0, 0];
+      frame.updates = null;
+      pushTransform(selectedElement.id, null);
+      // Inline start resetting has to be done very carefully here to avoid
+      // conflicts with stylesheets. See #3951.
+      target.style.transform = '';
+      target.style.width = '';
+      target.style.height = '';
+      if (moveable.current) {
+        moveable.current.updateRect();
+      }
+    },
+    [frame, pushTransform, selectedElement.id]
+  );
+
+  const canSnap =
+    !isEditMode && (!isDragging || (isDragging && !activeDropTargetId));
+  const hideHandles = isDragging || Boolean(draggingResource);
+
+  const classNames = classnames('default-moveable', {
+    'hide-handles': hideHandles,
+    'type-text': selectedElement.type === 'text',
+  });
+  const _dragProps = useDrag({
+    setIsDragging,
+    resetMoveable,
+    selectedElement,
+    setTransformStyle,
+    frame,
+  });
+  // No dragging in edit mode.
+  const dragProps = isEditMode
+    ? {
+        onDrag: () => false,
+        onDragEnd: () => false,
+        onDragStart: () => false,
+      }
+    : _dragProps;
```

Comment:
Was just thinking why we pass `isEditMode` for `useResize/useRotate` and not for `useDrag` (and do the "No dragging in edit mode." logic there), but I'm fine with this too.